lib/rules/curly.js

```diff
@@ -178,5 +237,8 @@ module.exports = function(context) {
 module.exports.schema = [
     {
         "enum": ["all", "multi", "multi-line", "multi-or-nest"]
+    },
+    {
+        "enum": ["consistent"]
```

Comment:
This will get slightly easier after #3625, but this should do the trick right now:

```json
{
    "anyOf": [
        {
            "type": "array",
            "items": [
                {
                    "enum": [0, 1, 2]
                },
                {
                    "enum": ["all"]
                }
            ],
            "minItems": 1,
            "maxItems": 2
        },
        {
            "type": "array",
            "items": [
                {
                    "enum": [0, 1, 2]
                },
                {
                    "enum": ["multi", "multi-line", "multi-or-nest"]
                },
                {
                    "enum": ["consistent"]
                }
            ],
            "minItems": 1,
            "maxItems": 3
        }
    ]
}
```